src/controllers/projectCtrl.js

```diff
@@ -37,66 +37,100 @@ exports.deleteProject = (req, res) => {
 };
 
 exports.updateProjectInfos = (req, res) => {
-  Project.findByIdAndUpdate(req.params.projectId, req.body, {
-    new: true,
-  })
+  //check if project exist
+  Project.findById(req.params.projectId)
     .then((project) => {
-      if (!project)
-        res
-          .status(400)
-          .json({ message: 'No Item updated: The project does not exist' });
-      else
-        res.status(200).json({
-          message: 'The project has been modified correctly!',
-          data: project,
+      if (project) {
+        project = {
+          ...project,
+          ...req.body,
+        };
+        project
+          .save()
+          .then((newProject) =>
+            res.status(200).json({
+              message: 'Project infos updated!',
+              newProject,
+            })
+          )
+          .catch((error) => res.status(500).json(error));
+      } else {
+        res.status(404).json({
+          message: 'Project not found!',
         });
+      }
```

Comment:
This is precisely the same as before, I think? Do you see a difference between both logic?

Reply:
You are wright on this function, I don't remember it was checking if project existed, maybe you changed it and I didn't notice and rebase